File: support/src/main/java/com/support/utills/ViewUtils.java
package com.support.utills;

import android.app.Activity;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Build;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.PopupWindow;

public class ViewUtils {
    private static String TAG = "ViewUtils";

    public static View getViewFromLayout(Activity mActivity, int mContainerID, int mLayoutID) {
        FrameLayout activityContainer = mActivity.findViewById(mContainerID);
        activityContainer.removeAllViews();
        View vi = mActivity.getLayoutInflater().inflate(mLayoutID, null);
        activityContainer.addView(vi);
        return activityContainer;
    }

    public static View getViewFromLayout(Activity mActivity, FrameLayout activityContainer, int mLayoutID) {
        activityContainer.removeAllViews();
        View vi = mActivity.getLayoutInflater().inflate(mLayoutID, null);
        activityContainer.addView(vi);
        return activityContainer;
    }

    public static View getViewFromLayout(Activity mActivity, int mLayoutID) {
        return mActivity.getLayoutInflater().inflate(mLayoutID, null, false);
    }

    public static void getSmoothAnimation(Activity activity) {
        activity.overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
    }

    public static void setViewAndChildrenEnabled(View view, boolean enabled) {
        view.setEnabled(enabled);
        if (view instanceof ViewGroup) {
            ViewGroup viewGroup = (ViewGroup) view;
            for (int i = 0; i < viewGroup.getChildCount(); i++) {
                View child = viewGroup.getChildAt(i);
                setViewAndChildrenEnabled(child, enabled);
            }
        }
    }

    public static PopupWindow getPopup(Activity mInstance, PopupWindow popupWindow, View contentView, View anchorView) {
        if (popupWindow == null) {
            popupWindow = new PopupWindow(mInstance);
        }
        popupWindow.setFocusable(true);
        popupWindow.setContentView(contentView);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            popupWindow.setElevation(5F);
        }
        popupWindow.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        popupWindow.setOutsideTouchable(true);
        popupWindow.showAsDropDown(anchorView); // where u want show on view click event popupwindow.showAsDropDown(view, x, y);
        return popupWindow;
    }
}

File: support/src/main/java/com/support/utills/ViewUtils.kt
package com.support.utills

import android.app.Activity
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Build
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.view.WindowManager
import android.widget.FrameLayout
import android.widget.PopupWindow
import androidx.annotation.RequiresApi
import com.google.android.material.internal.CheckableImageButton
import com.google.android.material.textfield.TextInputLayout
import com.support.R

object ViewUtils {
    private const val TAG = "ViewUtils"
    fun getViewFromLayout(mActivity: Activity, mContainerID: Int, mLayoutID: Int): View {
        val activityContainer = mActivity.findViewById<FrameLayout>(mContainerID)
        activityContainer.removeAllViews()
        val vi = mActivity.layoutInflater.inflate(mLayoutID, null)
        activityContainer.addView(vi)
        return activityContainer
    }

    fun getViewFromLayout(
        mActivity: Activity,
        activityContainer: FrameLayout,
        mLayoutID: Int
    ): View {
        activityContainer.removeAllViews()
        val vi = mActivity.layoutInflater.inflate(mLayoutID, null)
        activityContainer.addView(vi)
        return activityContainer
    }

    fun getViewFromLayout(mActivity: Activity, mLayoutID: Int): View {
        return mActivity.layoutInflater.inflate(mLayoutID, null, false)
    }

    fun getSmoothAnimation(activity: Activity) {
        activity.overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out)
    }

    fun setViewAndChildrenEnabled(view: View, enabled: Boolean) {
        view.isEnabled = enabled
        if (view is ViewGroup) {
            val viewGroup = view
            for (i in 0 until viewGroup.childCount) {
                val child = viewGroup.getChildAt(i)
                setViewAndChildrenEnabled(child, enabled)
            }
        }
    }

    fun getPopup(
        mInstance: Activity?,
        popupWindow: PopupWindow?,
        contentView: View?,
        anchorView: View?
    ): PopupWindow {
        var popupWindow = popupWindow
        if (popupWindow == null) {
            popupWindow = PopupWindow(mInstance)
        }
        popupWindow.isFocusable = true
        popupWindow.contentView = contentView
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            popupWindow.elevation = 5f
        }
        popupWindow.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        popupWindow.isOutsideTouchable = true
        popupWindow.showAsDropDown(anchorView) // where u want show on view click event popupwindow.showAsDropDown(view, x, y);
        return popupWindow
    }

    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
    fun setStatusBarColor(activity: Activity,color: Int){
        val window: Window = activity.window
        window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
        window.setStatusBarColor(color)
    }
}